Add brewing stand block and inventory support, minor fixes

MusicScore opened the pull request:
Fixes #40

MusicScore committed:
Add brewing stand block and inventory support, minor fixes

Add BrewingStandRefueled event

Add tag <InventoryTag.brewing_ingredient> and mechanism "InventoryTag.brewing_ingredient"

Add brewing stands to tag <InventoryTag.fuel> and mechanism "InventoryTag.fuel"

Reorganization of some InventoryTag code related to furnaces and crafting tables, added new helper methods for brewing stands

Add tags <LocationTag.brewing_time> and <LocationTag.brewing_fuel_level>, and mechanisms "LocationTag.brewing_time" and "LocationTag.brewing_fuel_level"

Fix the meta for mechanism "LocationTag.activate" due to lack of clarity

mergu commented on plugin/src/main/java/com/denizenscript/denizen/objects/InventoryTag.java:
```diff
@@ -668,6 +672,44 @@ public ItemTag getSmelting() {
+
+    public void setSmelting(ItemStack item) {
+        if (inventory instanceof FurnaceInventory) {
+            ((FurnaceInventory) inventory).setFuel(item);
```
Is this correct? Seems to be duplicate of the method above

mergu commented:
My only other suggestion for this would be a tag/mec pair for getting and setting the result slots too. Spigot doesn't provide that but you can reference them as inventory slots and set them. They should accept/output a list of length 3

MusicScore committed:
Fix BrewingStandRefueledScriptEvent, add new tag/mech pair

Allow the BrewingStandRefueled event to fire, add new contexts and fix determinations

Fix InventoryTag#setSmelting(ItemStack) and make use of this method

Fix meta typo for <InventoryTag.brewing_ingredient>

Add tag <InventoryTag.bottles> and mechanism "InventoryTag.bottles"

ghost commented:
Fix implemented, new tag/mec pair added (`<InventoryTag.bottles>` and `InventoryTag.bottles`).

Tested to work on Paper 1.14.4.

EDIT: Changed "bottles" to "brewing_results".

MusicScore committed:
Rename "InventoryTag.bottles" tag/mechanism to "InventoryTag.brewing_results"
